cli/src/lib.rs

```diff
@@ -130,6 +132,26 @@ impl NodeBuilder {
         vec![InstanceCollection::from(Supervisor)]
     }
 
+    /// Adds a new Runtime to the list of available runtimes.
+    ///
+    /// Note that you don't have to add a Rust Runtime, since it's included by default.
+    pub fn with_external_runtime(mut self, runtime: impl Into<(u32, Box<dyn Runtime>)>) -> Self {
+        self.external_runtimes.push(runtime.into());
+        self
+    }
+
+    /// Adds a new InstanceCollection to the list of available services.
+    ///
+    /// With that method you can start Exonum with already deployed & running service
+    /// instances.
+    ///
+    /// Note that if you add the `InstanceCollection` manually, you don't need to call
+    /// `with_service`, since `InstanceCollection` already contains a service inside.
+    pub fn with_instance_collection(mut self, instance_collection: InstanceCollection) -> Self {
```

Comment:
I mean that we should provide runtime-agnostic mechanism to add instances at startup.

Something like this:

```rust
struct Instance {
    name: String,
    artifact: ArtifactId,
    constructor: Vec<u8>
}
```

Reply:
Ah. I see. I thought you need it as a temporary solution to run services without `exonum-launcher`.

I'll remove it then.

Reply:
However, runtime-independent deploy at the start is much bigger question, and totally out of the scope of this PR.